### PR TITLE
maintain transferRequests state

### PR DESCRIPTION
**Pull request by neerk28**

### CODE REVIEW

File: functions/api/action.js [hunk 1]
```diff
@@ -309,28 +327,42 @@ router.post('/rejectRequest', async (req, res, next) => {
...
+        accepted: false
+    }
+    const fromBranchData = {
+        ...fromBranchPendingReqDoc.data(),
+        id: fromBranchPendingReqDoc.id,
+        accepted: false
+    }
+    //fire and forget request state tracker
+    toBranchRef.collection("requestsHistory").add(toBranchData)
+    fromBranchRef.collection("requestsHistory").add(fromBranchData)
```
Comment (ramyarv) [inline functions/api/action.js]: Is this the new collection for maintaining whether the pending request was accepted or rejected ?

Comment (ramyarv) [reply functions/api/action.js]: One more thing, are we going to get only the rejected requests stored here ? And how about the accepted requests @neerk28 ?

**Review by ramyarv — COMMENTED** [review]
Have some queries

File: functions/api/action.js (hunk 1 above)
Comment (shrikantdange1208) [reply functions/api/action.js]: @ramyarv Both are saved in this collection. There is a field accepted (true/false).

File: functions/api/action.js [hunk 2]
```diff
@@ -225,8 +225,26 @@ router.post('/transferProduct', async (req, res, next) => {
...
-    if(toBranchPendingReqDoc.exists) await toBranchPendingReqRef.delete()
-    if(fromBranchPendingReqDoc.exists) await fromBranchPendingReqRef.delete()
+    if (toBranchPendingReqDoc.exists) {
+        const toBranchData = {
+            ...toBranchPendingReqDoc.data(),
+            id: toBranchPendingReqDoc.id,
+            accepted: true
+        }
+        //fire and forget request state tracker
+        toBranchRef.collection("requestsHistory").add(toBranchData)
```
Comment (shrikantdange1208) [inline functions/api/action.js]: Could you please add a constant for requestsHistory?

**Review by shrikantdange1208 — APPROVED** [review]
LGTM. Only the constant for requestsHistory

File: functions/api/action.js (hunk 1 above)
Comment (ramyarv) [reply functions/api/action.js]: sure @shrikantdange1208  . just curious to know, we could have added the status of request in the pending requests itself right like PENDING/ACCEPTED/REJECTED. why we need a separate collection for it ?

Comment (neerk28) [reply functions/api/action.js]: we could do that as well, its just a design choice. in get of pending requests, UI should pass a filter for state: PENDING to show in inventory screen, and for dashboard they could pass state:ACCEPTED/REJECTED.

**Review by shrikantdange1208 — APPROVED** [review]
LGTM

**Pull request merged**